src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,68 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.commands.Command;
+import duke.exceptions.DukeException;
+import duke.exceptions.NoSuchTaskException;
+import duke.tasks.TaskList;
+
+
+/**
+ * Encapsulates the information for a Duke object that contains a Storage, Parser and TaskList object.
+ */
+public class Duke {
+    private TaskList items;
+    private final Storage storage;
+    private final Parser parser;
+
+    /**
+     * Constructs a Duke object with the specified file path for the data file.
+     */
+    public Duke() {
+        this.items = new TaskList();
+        this.storage = new Storage("data/duke.txt");
+        this.parser = new Parser();
+        this.start();
+    }
+
+    /**
+     * Starts the chat bot by attempting to load the previously stored data.
+     * Chat bot will receive, interpret, execute and save data from the recognisable commands from user.
+     */
+    private void start() {
+        try {
+            this.items = this.storage.loadTask();
+        } catch (IOException | NoSuchTaskException e) {
+            Ui.notifyError(e.getMessage());
+        }
+    }
+
+    public String getResponse(String input) {
+        try {
+            String exitTag = "bye";
+            if (input.equalsIgnoreCase(exitTag)) {
+                new Thread() {
+                    public void run() {
+                        try {
+                            Thread.sleep(1250);
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                        } finally {
+                            System.exit(0);
+                        }
+                    }
+                }.start();
+
```

Comment:
There is some deep nesting here, maybe you can try simplifying it to be less "arrow-head" like.

Reply:
Not really a coding violation, but I do agree with Siddanth. Improving readability here would be nice.

Reply:
Thank you. Will change this part in the next commit.